cdap-api/src/main/java/co/cask/cdap/api/metadata/MetadataEntity.java

```diff
@@ -277,6 +277,110 @@ public static MetadataEntity ofDataset(String namespace, String datasetName) {
     return subParts;
   }
 
+  /**
+   * @return a {@link String} which describes the {@link MetadataEntity} for a user in plain english.
+   * If the {@link MetadataEntity} represents a known CDAP entity then the description is worded to show relations
+   * in the hierarchy.
+   */
+  public String getDescription() {
+    return getDescription(new StringBuilder(), getType());
+  }
+
+  private String getDescription(StringBuilder builder, String type) {
```

Comment:
It seems like it would be more readable code if you used `String.format` and String concatenation where necessary, instead of using StringBuilder like you are currently doing.
The performance won't be as good, but it doesn't seem like we need to make `getDescription` super performant.

Reply:
Agreed it will make it much more readable. Yes, we dont accept this call to be infrequent.